tesseract-server/src/handlers/aggregate.rs

```diff
@@ -60,6 +48,15 @@ pub fn do_aggregate(
 {
     let (cube, format) = cube_format;
 
+    // Get cube object to check for API key
+    let schema = &req.state().schema.read().unwrap().clone();
+    let cube_obj = ok_or_404!(schema.get_cube_by_name(&cube));
```

Comment:
once the schema/cube_obj are unpacked here, can we reuse this for the  `..read().unwrap()..` that`query_ir_headers` does now?